src/vcpkg/binarycaching.cpp

```diff
@@ -31,42 +31,44 @@ namespace
-                                 (),
-                                 "",
-                                 "unknown binary provider type: valid providers are 'clear', 'default', 'nuget', "
-                                 "'nugetconfig', 'interactive', and 'files'");
-    DECLARE_AND_REGISTER_MESSAGE(InvalidArgumentRequiresPrefix,
-                                 (msg::binary_source),
+    DECLARE_AND_REGISTER_MESSAGE(
+        AuthenticationMayRequireManualAction,
+        (msg::vendor),
+        "",
+        "One or more {vendor} credential providers requested manual action. Add the binary source "
+        "'interactive' to allow interactivity.");
+    DECLARE_AND_REGISTER_MESSAGE(CompressFolderFailed, (msg::path), "", "Failed to compress folder '{path}':");
+    DECLARE_AND_REGISTER_MESSAGE(
+        DefaultPathToBinaries,
+        (msg::path),
+        "",
+        "Based on your system settings, the default path to store binaries is \n '{path}'. This consults "
+        "%LOCALAPPDATA%/%APPDATA% on Windows and $XDG_CACHE_HOME or $HOME on other platforms.");
+    DECLARE_AND_REGISTER_MESSAGE(ExtendedDocumenationAtUrl,
+                                 (msg::url),
                                  "",
-                                 "invalid argument: binary config '{binary_source}' requires at least a prefix");
-    DECLARE_AND_REGISTER_MESSAGE(InvalidArgumentRequiresSource,
-                                 (msg::binary_source),
+                                 "Extendend documentation available at '{url}'.");
+    DECLARE_AND_REGISTER_MESSAGE(FailedToStoreBinaryCache,
+                                 (msg::path, msg::error_msg),
                                  "",
-                                 "invalid argument: binary config '{binary_source}' requires non-empty source");
-    DECLARE_AND_REGISTER_MESSAGE(InvalidArgument, (), "", "invalid argument");
+                                 "Failed to store binary cache '{path}':'{error_msg}'");
     DECLARE_AND_REGISTER_MESSAGE(
-        InvalidArgumentRequiresSourceArgument,
-        (msg::binary_source),
+        FailedVendorAuthentication,
+        (msg::vendor, msg::url),
         "",
-        "invalid argument: binary config '{binary_source}' requires at least a source argument");
-    DECLARE_AND_REGISTER_MESSAGE(InvalidArgumentRequiresPathArgument,
-                                 (msg::binary_source),
-                                 "",
-                                 "invalid argument: binary config '{binary_source}' requires at least a path argument");
+        "One or more {vendor} credential providers failed to authenticate. See '{url}' for more details"
+        "on how to provide credentials.");
+    DECLARE_AND_REGISTER_MESSAGE(HelpTableFormat,
+                                 (msg::value),
+                                 "'{value}' is vcpkg::HelpTableFormatter as a string.",
+                                 "'{value}'");
```

Comment:
This seems wrong: The old code just printed the thing, this is adding extra 's around it?

It seems like the thing itself needs to be localized rather than being washed through format like this.